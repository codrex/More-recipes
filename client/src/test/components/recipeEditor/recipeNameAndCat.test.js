/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import toJSON from 'enzyme-to-json';
import {
  PureRecipeNameAndCategory
} from '../../../components/pages/RecipeEditor/RecipeNameAndCategory';


const props = {
  handleSubmit: jest.fn(),
  clearValidationError: jest.fn(),
  recipe: {},
  updateName: jest.fn(),
  updateCategory: jest.fn(),
  validationError: {},
  name: '',
  loading: false,
};

describe('Recipe name and category component', () => {
  let component;
  beforeEach(() => {
    component = shallow(<PureRecipeNameAndCategory {...props} />);
  });
  test('expected to match snapshot', () => {
    const tree = toJSON(component);
    expect(tree).toMatchSnapshot();
  });

  test('expected to match snapshot when name field has focus', () => {
    component.find('Form').find('#recipeName').simulate('focus');
    const tree = toJSON(component);
    expect(tree).toMatchSnapshot();
  });

  test('expected to match snapshot when name field is blurred', () => {
    component.find('Form').find('#recipeName').simulate('blur');
    const tree = toJSON(component);
    expect(tree).toMatchSnapshot();
  });
});

