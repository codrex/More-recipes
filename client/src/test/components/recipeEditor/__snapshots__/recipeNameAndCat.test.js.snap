// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe name and category component expected to match snapshot 1`] = `
<div>
  <h4
    className="items-header-text"
  >
    Recipe name and category
  </h4>
  <Form
    action=""
    className=""
    disabled={false}
    id="submit"
    lg={true}
    method=""
    onSubmit={[Function]}
    primary={false}
    primaryInverse={false}
    secondary={false}
    secondaryInverse={false}
    style={
      Object {
        "zIndex": 4,
      }
    }
    submitBtnText=""
  >
    <Field
      className="add-recipe-input"
      component={[Function]}
      externalError={
        Object {
          "error": undefined,
          "touched": true,
        }
      }
      id="recipeName"
      name="name"
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="Enter recipe name"
      type="text"
      validate={[Function]}
    />
    <Select
      className="add-recipe-input"
      externalError={
        Object {
          "error": undefined,
          "touched": true,
        }
      }
      fgClassName=""
      handleChange={[Function]}
      helpTextClassName=""
      id="category"
      input={Object {}}
      meta={Object {}}
      name="category"
      options={
        Array [
          Object {
            "label": "Breakfast",
            "value": "breakfast",
          },
          Object {
            "label": "Brunch",
            "value": "brunch",
          },
          Object {
            "label": "Lunch",
            "value": "lunch",
          },
          Object {
            "label": "Snacks",
            "value": "snacks",
          },
          Object {
            "label": "Appertisers",
            "value": "appertisers",
          },
          Object {
            "label": "Dinner",
            "value": "dinner",
          },
          Object {
            "label": "Soup",
            "value": "soup",
          },
          Object {
            "label": "Noodles",
            "value": "noodles",
          },
          Object {
            "label": "Vegeterian",
            "value": "vegeterian",
          },
          Object {
            "label": "Salads",
            "value": "salads",
          },
          Object {
            "label": "Sides",
            "value": "sides",
          },
          Object {
            "label": "Sauces",
            "value": "sauces",
          },
          Object {
            "label": "Baking",
            "value": "baking",
          },
          Object {
            "label": "Desserts",
            "value": "desserts",
          },
          Object {
            "label": "Drinks",
            "value": "drinks",
          },
          Object {
            "label": "Cookies and Candy",
            "value": "cookies and candy",
          },
          Object {
            "label": "Miscellaneous",
            "value": "miscellaneous",
          },
        ]
      }
      placeholder="Select recipe category"
      type="text"
      value=""
    />
  </Form>
</div>
`;

exports[`Recipe name and category component expected to match snapshot when name field has focus 1`] = `
<div>
  <h4
    className="items-header-text"
  >
    Recipe name and category
  </h4>
  <Form
    action=""
    className=""
    disabled={false}
    id="submit"
    lg={true}
    method=""
    onSubmit={[Function]}
    primary={false}
    primaryInverse={false}
    secondary={false}
    secondaryInverse={false}
    style={
      Object {
        "zIndex": 4,
      }
    }
    submitBtnText=""
  >
    <Field
      className="add-recipe-input"
      component={[Function]}
      externalError={
        Object {
          "error": undefined,
          "touched": true,
        }
      }
      id="recipeName"
      name="name"
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="Enter recipe name"
      type="text"
      validate={[Function]}
    />
    <Select
      className="add-recipe-input"
      externalError={
        Object {
          "error": undefined,
          "touched": true,
        }
      }
      fgClassName=""
      handleChange={[Function]}
      helpTextClassName=""
      id="category"
      input={Object {}}
      meta={Object {}}
      name="category"
      options={
        Array [
          Object {
            "label": "Breakfast",
            "value": "breakfast",
          },
          Object {
            "label": "Brunch",
            "value": "brunch",
          },
          Object {
            "label": "Lunch",
            "value": "lunch",
          },
          Object {
            "label": "Snacks",
            "value": "snacks",
          },
          Object {
            "label": "Appertisers",
            "value": "appertisers",
          },
          Object {
            "label": "Dinner",
            "value": "dinner",
          },
          Object {
            "label": "Soup",
            "value": "soup",
          },
          Object {
            "label": "Noodles",
            "value": "noodles",
          },
          Object {
            "label": "Vegeterian",
            "value": "vegeterian",
          },
          Object {
            "label": "Salads",
            "value": "salads",
          },
          Object {
            "label": "Sides",
            "value": "sides",
          },
          Object {
            "label": "Sauces",
            "value": "sauces",
          },
          Object {
            "label": "Baking",
            "value": "baking",
          },
          Object {
            "label": "Desserts",
            "value": "desserts",
          },
          Object {
            "label": "Drinks",
            "value": "drinks",
          },
          Object {
            "label": "Cookies and Candy",
            "value": "cookies and candy",
          },
          Object {
            "label": "Miscellaneous",
            "value": "miscellaneous",
          },
        ]
      }
      placeholder="Select recipe category"
      type="text"
      value=""
    />
  </Form>
</div>
`;

exports[`Recipe name and category component expected to match snapshot when name field is blurred 1`] = `
<div>
  <h4
    className="items-header-text"
  >
    Recipe name and category
  </h4>
  <Form
    action=""
    className=""
    disabled={false}
    id="submit"
    lg={true}
    method=""
    onSubmit={[Function]}
    primary={false}
    primaryInverse={false}
    secondary={false}
    secondaryInverse={false}
    style={
      Object {
        "zIndex": 4,
      }
    }
    submitBtnText=""
  >
    <Field
      className="add-recipe-input"
      component={[Function]}
      externalError={
        Object {
          "error": undefined,
          "touched": true,
        }
      }
      id="recipeName"
      name="name"
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="Enter recipe name"
      type="text"
      validate={[Function]}
    />
    <Select
      className="add-recipe-input"
      externalError={
        Object {
          "error": undefined,
          "touched": true,
        }
      }
      fgClassName=""
      handleChange={[Function]}
      helpTextClassName=""
      id="category"
      input={Object {}}
      meta={Object {}}
      name="category"
      options={
        Array [
          Object {
            "label": "Breakfast",
            "value": "breakfast",
          },
          Object {
            "label": "Brunch",
            "value": "brunch",
          },
          Object {
            "label": "Lunch",
            "value": "lunch",
          },
          Object {
            "label": "Snacks",
            "value": "snacks",
          },
          Object {
            "label": "Appertisers",
            "value": "appertisers",
          },
          Object {
            "label": "Dinner",
            "value": "dinner",
          },
          Object {
            "label": "Soup",
            "value": "soup",
          },
          Object {
            "label": "Noodles",
            "value": "noodles",
          },
          Object {
            "label": "Vegeterian",
            "value": "vegeterian",
          },
          Object {
            "label": "Salads",
            "value": "salads",
          },
          Object {
            "label": "Sides",
            "value": "sides",
          },
          Object {
            "label": "Sauces",
            "value": "sauces",
          },
          Object {
            "label": "Baking",
            "value": "baking",
          },
          Object {
            "label": "Desserts",
            "value": "desserts",
          },
          Object {
            "label": "Drinks",
            "value": "drinks",
          },
          Object {
            "label": "Cookies and Candy",
            "value": "cookies and candy",
          },
          Object {
            "label": "Miscellaneous",
            "value": "miscellaneous",
          },
        ]
      }
      placeholder="Select recipe category"
      type="text"
      value=""
    />
  </Form>
</div>
`;
